homeassistant/components/matter/number.py

```diff
@@ -405,4 +406,21 @@ def _update_from_device(self) -> None:
             custom_clusters.InovelliCluster.Attributes.LEDIndicatorIntensityOn,
         ),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.NUMBER,
+        entity_description=MatterNumberEntityDescription(
+            key="speaker_setpoint",
+            translation_key="speaker_setpoint",
+            native_unit_of_measurement=PERCENTAGE,
+            native_max_value=100,
+            native_min_value=0,
+            native_step=1,
+            device_to_ha=lambda x: round(x / 2.55),  # Matter range (0-255)
```

Comment:
Matter is 0..254 or 1..254, so there is no 255 Ideally we should use the minLevel and maxLevel (fallback if not existing=254) attributes for the calculation

Reply:
Also: depending on minLevel the value of "0" might be not allowed!